ap_src/ap_app/static/js/teams.js

```diff
@@ -53,26 +53,47 @@ function favouriteTeam(e, user, id) {
     })
 }
 
-function LeaveTeam(e, user, redirect) {
-    var data = JSON.stringify({"username": user, "team": e.id})
-    fetch(apiURL + 'api/manage/?method=leave', {
-        method: "post",
-        body: data,
-        headers: {
-            "Content-Type": "application/json",
+/*
+Leaves team but also checks for lingering permissions and removes them
+*/
+async function LeaveTeam(e, user, token) {
```

Comment:
You could change the function name to something like LeaveTeamAndRemovePermissions to be more descriptive of what it's doing.

Or the way I would implement it is to have a function called handleLeaveButtonClick, and then in handleLeaveButtonClick I'd call a function called leaveTeam, and then another called checkLingeringPermissions, and then another called removeLingeringPermissions. Something along those lines.

Some reading on why if you're interested: [Functions should do exactly one thing](https://medium.com/@saddamhr/clean-code-functions-should-do-exactly-one-thing-8bf648bc71ea)

Reply:
I think for now, changing the name would be fine, but this could be made a separate issue for the future.

Reply:
The name was changed with 36a6a06